### PR TITLE
Install essential requirements with its own order

### PR DESCRIPTION
**Pull request by happz**
It turns out essential requirements must be installed before regular phases start running, because these phases may need their requirements to even begin. In particular, we noticed `prepare/ansible` requires a Python interpreter, but the default `order` puts it before installation of (collected) requirements.

Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

**Comment by happz** [comment]
@psss attempt to address the essential requirement conundrum. I left out the documentation on purpose: I think various special `order` values deserve their own section, hopefully in 1.39, and this would fit in.

**Comment by lukaszachy** [comment]
` /tests/try/basic` is failing:
- no /usr/bin/python for  './epel.exp'  (shouldn't that be solved by this PR?)
- changed messages for './install.exp' (f-41 got released, is that it?)

File: tmt/steps/prepare/__init__.py [hunk 1]
```diff
@@ -100,6 +100,34 @@ def go(
...
 
 
+# Required & recommended packages
+#
+# Structures and code for collecting requirements for different guests
+# or their groups to avoid installing all test requirements on all
+# guests. For example, a test running on the "server" guest might
+# require package `foo` while the test running on the "client" might
+# require package `bar`, and `foo` and `bar` cannot be installed at the
+# sametime.
```
Comment (psss) [inline tmt/steps/prepare/__init__.py:110]:
```suggestion
# same time.
```

**Review by psss — APPROVED** [review]
Thanks for fixing this so fast! Looks good, just one typo, not sure it's worth the respin.

File: tmt/steps/prepare/__init__.py (hunk 1 above)
Comment (psss) [reply tmt/steps/prepare/__init__.py:110]: Let's not block on that. Fixed in #3337.

**Comment by psss** [comment]
Reviewed failures, all are related to the `fedora-41` changes.

**Pull request merged**